### ISSUE TITLE
Error during large dataset creation

### ISSUE THREAD

**Issue by ShaktiKShrivastava** [rodrigopivi/Chatito#17]
Hi,

I included a range of numbers in one of the synonyms block, and the page stops responding. When I replaced them with a single constant string, it works.

This does not work,

%[lookForSomething]
    ~[find?] ~[all?] ~[free?] ~[restaurants?] from @[bookTime] to @[bookTime]
    ~[find?] ~[all?] ~[free?] ~[restaurants?]
    ~[find?] ~[all?] ~[free?] ~[restaurants?] from @[bookTime] to @[bookTime] on @[bookDate]

@[bookTime#snips/datetime]
    ~[someDigit] ~[am?]
    ~[someDigit] ~[pm?]
    ~[someDigit] ~[:] ~[someDigit] ~[am?]
    ~[someDigit] ~[:] ~[someDigit] ~[pm?]

@[bookDate#snips/datetime]
    today
    tomorrow
    day after tomorrow
    10th March
    1st August
    12th July

~[find]
    show
    look for
    search
    show me
    find
    identify

~[all]
    any

~[free]
    unoccupied

~[rooms]
    room

~[:]
    :

~[am]
    am

~[pm]
    pm

~[someDigit]
	1
	2
	3
	4
	5
	6
	7
	8
	9

**However, when I remove the ~[someDigit] synonyms (which largely reduces the number of combinations which Chatito has to generate) from the entity definitions, it works,**

%[lookForSomething]
    ~[find?] ~[all?] ~[free?] ~[restaurants?] from @[bookTime] to @[bookTime]
    ~[find?] ~[all?] ~[free?] ~[restaurants?]
    ~[find?] ~[all?] ~[free?] ~[restaurants?] from @[bookTime] to @[bookTime] on @[bookDate]

@[bookTime#snips/datetime]
    someDigit ~[am?]
    someDigit ~[pm?]
    someDigit ~[:] someDigit ~[am?]
    someDigit ~[:] someDigit ~[pm?]

@[bookDate#snips/datetime]
    today
    tomorrow
    day after tomorrow
    10th March
    1st August
    12th July

~[find]
    show
    look for
    search
    show me
    find
    identify

~[all]
    any

~[free]
    unoccupied

~[rooms]
    room

~[:]
    :
~[am]
    am

~[pm]
    pm

Any thoughts on this, or any place I am going wrong?

**Comment by ShaktiKShrivastava** [comment]
Also Get the following error in the first case,
`ERROR: 
 {"line": 12, "column": 45}`

**Comment by rodrigopivi** [comment]
well you can youse the npm package, it should work faster, but still, chatito generates all possible combinations, that is why when you add more variations it takes much longer too, im in the process of re-implementing the core so we can just generate the needed samples on demand, stay tuned

**Comment by ShaktiKShrivastava** [comment]
Yeah, I was surprised with the program throwing out errors for large datasets, instead of going unresponsive. Probably I should have named this issue as 'Error during large dataset creation'

**Comment by hadils** [comment]
I am using npm on Mac OS X 10.13.2. I am still new to Chatito and Rasa. I get the following error:
```
FULL ERROR REPORT:
RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at Object.<anonymous> (/Users/hadilsabbagh/node_modules/chatito/bin/chatito.js:50:49)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Function.Module.runMain (module.js:665:10)
    at findNodeScript.then.existing (/usr/local/lib/node_modules/npm/node_modules/libnpx/index.js:268:14)
    at <anonymous>
```

With the following input file:
```
%[greet]
    ~[greeting] ~[botName?]

%[open_application]
    ~[please?] ~[open] @[application]

%[parallel_open_application]
    ~[parallel] ~[please?] ~[open] @[application]

%[xor_open_application]
    ~[xor_conditional] @[condition] ~[please?] ~[open] @[application]

~[parallel]
    In parallel
    Meanwhile
    at the same time
    in the meantime

~[xor_conditional]
    if
    whenever
    otherwise
    otherwise if

@[condition]
    a condition is met
    a condition is not met
    a task is completed
    a task is not completed
    a problem is resolved
    a problem is not resolved
    something is desired
    something is not desired
    something else is desired
    something else is not desired

~[greeting]
    hi
    hey
    hello
    greetings
    howdy
    yo

~[botName]
    Xeia
    xeia
    XEIA

~[please]
    please
    pls

~[open]
    open
    start up
    check

@[application]
    browser
    Chrome
    IE
    Edge
    Firefox
    Mozilla
    Internet Explorer
    Safari
    web browser
    Word
    Excel
    PowerPoint
    powerpoint
    Google Docs
    Google Sheets
    Google Slides
    editor
    application
    Adobe
    email
    Mail
    Mail application
    Outlook

%[navigate_browser]
    ~[please?] ~[goto] @[website] ~[website_modifiers?]
    ~[please?] ~[goto] the @[website] ~[website_modifiers?]

%[parallel_navigate_browser]
    ~[parallel] ~[please?] ~[goto] @[website] ~[website_modifiers?]
    ~[parallel] ~[please?] ~[goto] the @[website] ~[website_modifiers?]

%[xor_navigate_browser]
    ~[xor_conditional] @[condition] ~[please?] ~[goto] @[website] ~[website_modifiers?]
    ~[xor_conditional] @[condition] ~[please?] ~[goto] the @[website] ~[website_modifiers?]

~[goto]
    go to
    navigate to
    find
    search for

@[website]
    www.google.com
    teamworks.wellsfargo.com
    www.wellsfargo.com
    Wells Fargo
    Google
    American Airlines
    aaa.com
    xeia.io
    xeia.ai
    Xeia
    XEIA

~[website_modifiers]
    website
    web page

%[edit_document]
    ~[please?] ~[edit] @[document]
    ~[please?] ~[edit] the @[document]
    ~[please?] ~[edit] the @[time1] @[document]
    ~[please?] ~[edit] the @[document] @[time2]

%[parallel_edit_document]
    ~[parallel] ~[please?] ~[edit] @[document]
    ~[parallel] ~[please?] ~[edit] the @[document]
    ~[parallel] ~[please?] ~[edit] the @[time1] @[document]
    ~[parallel] ~[please?] ~[edit] the @[document] @[time2]

%[xor_edit_document]
    ~[xor_conditional] @[condition] ~[please?] ~[edit] @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[edit] the @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[edit] the @[time1] @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[edit] the @[document] @[time2]

~[edit]
    open
    edit
    append to
    create
    create a new document named
    write to
    display

@[time1]
    latest
    earliest
    most recent
    oldest
    youngest

@[time2]
    created on Thursday, June 7, 2018
    created on 6/7/18
    created today at 2 pm
    last modified on Thursday, June 7, 2018
    last modified on 6/7/18
    last modified today at 2 pm
    from September 11, 2001
    from 2001
    from 9/11/2011
    from 2 pm
    last updated on September 11, 2001
    last updated in 2001
    last updated 2001
    last updated at 2 pm

@[document]
    abc.txt
    file
    Word document
    Excel spreadsheet
    Powerpoint presentation
    document
    spreadsheet
    presentation
    dissertation
    web page
    persons document
    persons spreadsheet
    persons web page
    persons presentation
    persons file
    document written by person
    spreadsheet written by person
    presentation written by person

%[delete_document]
    ~[please?] ~[delete] @[document]
    ~[please?] ~[delete] the @[document]
    ~[please?] ~[delete] the @[time1] @[document]
    ~[please?] ~[delete] the @[document] @[time2]

%[parallel_delete_document]
    ~[parallel] ~[please?] ~[delete] @[document]
    ~[parallel] ~[please?] ~[delete] the @[document]
    ~[parallel] ~[please?] ~[delete] the @[time1] @[document]
    ~[parallel] ~[please?] ~[delete the @[document] @[time2]

%[xor_delete_document]
    ~[xor_conditional] @[condition] ~[please?] ~[delete] @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[delete] the @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[delete] the @[time1] @[document]
    ~[xor_conditional] @[condition] ~[please?] ~[delete the @[document] @[time2]

~[delete]
    delete
    remove
    destroy
    wipe out
    rm
    junk
```

**Comment by rodrigopivi** [comment]
this is now fixed in version 2.0.0, there is an example with tests similar to this that works:
https://github.com/rodrigopivi/Chatito/blob/master/examples/dateBooking_large.chatito
 please read the docs since there were some chantes
